packages/account-v2/src/context/SignupWizardContext.tsx

```diff
@@ -47,24 +71,54 @@ export const useSignupWizardContext = () => {
     return context;
 };
 
+function valuesReducer(state: TState, action: TActions) {
+    const { payload, type } = action;
+    switch (type) {
+        case ACTION_TYPES.SET_CURRENCY:
+            return {
+                ...state,
+                currency: payload.currency,
+            };
+        case ACTION_TYPES.SET_PERSONAL_DETAILS:
+            return {
+                ...state,
+                firstName: payload.firstName,
+                lastName: payload.lastName,
+            };
```

Comment:
@shahzaib-deriv cause you have payload with needed fields you can write like this for all action types:

```suggestion
        case ACTION_TYPES.SET_CURRENCY:
        case ACTION_TYPES.SET_PERSONAL_DETAILS:
            return {
                ...state,
                ...payload
            };
```

Reply:
Thanks @sergei-deriv for the suggestion, WIll update when we'll add more actions